### DISCUSSION TITLE
Confirmation before applying file operations

### DISCUSSION THREAD

**Discussion by Ghaydn** [godotengine/godot-proposals#8043] [Editor]
Recently I was working on my project and accidently dragged 'scenes' folder (in FileSystem in the editor) to another folder. Godot immediately moved it and tried to save all opened scenes and project; after that it showed popup, that not all scenes can be properly saved. I tried to ctrl+z that, but it didn't work. Then I moved my folder back to where it was. Everything saved again (kinda) and then crashed.
Moving files and changing all their dependences is a major change. It may break some scenes or the whole project. Moreover, it cannot be simply undone. User must not be able to do that by accident.

What I suggest.
Add a setting (in editor settings) with 3 options:
1. Always move files when dragging in FileSystem (in case if someone does that really often)
2. Show confirmation popup (default)
3. Never drag files (in case if someone never does that), print in console if attempted.

**Comment by Calinou** [comment]
Thanks for the proposal! Consolidating in https://github.com/godotengine/godot-proposals/issues/7862.
